shm_wrapper/shm_wrapper.c

```diff
@@ -64,28 +80,21 @@ static void print_bitmap (int num_bits, uint32_t bitmap)
 	printf("\n");
 }
 
-//a few very useful semaphore operation wrapper utilities
-static void my_sem_wait (sem_t *sem, char *sem_desc)
-{
-	if (sem_wait(sem) == -1) {
-		log_printf(ERROR, "sem_wait: %s. %s", sem_desc, strerror(errno));
-	}
-}
-
-static void my_sem_post (sem_t *sem, char *sem_desc)
-{
-	if (sem_post(sem) == -1) {
-		log_printf(ERROR, "sem_post: %s. %s", sem_desc, strerror(errno));
-	}
-}
-
-static void my_sem_close (sem_t *sem, char *sem_desc)
+//function that returns dev_ix of specified device if it exists (-1 if it doesn't)
+static int get_dev_ix_from_uid (uint64_t dev_uid)
```

Comment:
Actually I think executor will need to use this function so can you make it public and add it to the header.

Reply:
What will you use it for?
Although I guess since it is a generally useful utility now, I can expose this as part of the API too